lldb/source/Core/ValueObjectDynamicValue.cpp

```diff
@@ -443,7 +443,7 @@ bool ValueObjectDynamicValue::DynamicValueTypeInfoNeedsUpdate() {
   auto *cached_ctx = m_value.GetCompilerType().GetTypeSystem();
   llvm::Optional<SwiftASTContextReader> scratch_ctx =
       GetScratchSwiftASTContext();
-  return scratch_ctx ? (cached_ctx == scratch_ctx->get()) : !cached_ctx;
+  return scratch_ctx ? (cached_ctx != scratch_ctx->get()) : !cached_ctx;
```

Comment:
can we get rid of the ternary, to hopefully focus on the important part?

I think there might technically be another logic bug:

If `cached_ctx` is null, and somehow `scratch_ctx` is also null, then the comparison `cached_ctx != scratch_ctx->get()` would then be `false`, which is saying no update is needed. Given the asserts, it should never happen, but you never know.

If we agree this is a logic bug, then what about?

```cpp
if (!scratch_ctx || !cached_ctx)
  return true;
return cached_ctx != scratch_ctx->get();
```

Reply:
Not entirely sure what the right behaviour is here, but I agree that we should get rid of the ternary op.